src/fs_clonefile.cpp

```diff
@@ -67,24 +67,24 @@ writen(const int     fd,
           return -1;
         }
 
-      nleft -= nwritten;
-      buf   += nwritten;
+      nleft -= (size_t)nwritten;
+      buf   += (size_t)nwritten;
     }
 
-  return count;
+  return nwritten;
```

Comment:
It shouldn't, no. But while making hfsinspect I ran into this issue a few times.  There was some kind of situation where the read returned early but I wasn't at the end of the file/device yet and the next read kept going.  So, since then, I've just always returned the real count so that client apps' counters are accurate and can handle that.